plugins/synaptics-rmi/README.md

```diff
@@ -0,0 +1,37 @@
+Synaptics RMI4 Support
+======================
+
+Introduction
+------------
+
+This plugin updates integrated Synaptics RMI4 devices, typically touchpads.
+
+GUID Generation
+---------------
+
+The HID DeviceInstanceId values are used, e.g. `HIDRAW\VEN_06CB&DEV_4875`.
+
+These devices also use custom GUID values constructed using the board ID, e.g.
+
+ * `SYNAPTICS_RMI\TM3038-002`
+
+Firmware Format
+---------------
+
+The daemon will decompress the cabinet archive and extract a firmware blob in
+a proprietary (but docucumented) file format.
+
+This plugin supports the following protocol ID:
+
+ * com.synaptics.rmi
+
+Fuzzing
+-------
+
+    CC=afl-gcc meson --default-library=static ../
+    AFL_HARDEN=1 ninja
+    mkdir rmi-fuzzing
+    ./plugins/synaptics-rmi/synaptics-rmi-dump gen0x rmi-fuzzing/example0x.img
+    ./plugins/synaptics-rmi/synaptics-rmi-dump gen10 rmi-fuzzing/example10.img
+    ../contrib/afl-fuzz.py -i rmi-fuzzing -o rmi-findings \
```

Comment:
Shouldn't this be `./contrib/afl-fuzz.py -i rmi-fuzzing -o rmi-findings \` (IE one period)

Reply:
You need to be in 'build' rather than in the project root. I do wonder if we should hook these up as meson targets tho.

Reply:
I did the meson target thing. I think it's much less likely to get broken accidentally.